apollo-federation/src/query_graph/mod.rs

```diff
@@ -548,6 +565,26 @@ impl QueryGraph {
             })
     }
 
+    pub(crate) fn context_id_by_source_and_argument(
+        &self,
+        source: &str,
+        argument: &ObjectFieldArgumentDefinitionPosition,
+    ) -> Result<&Name, FederationError> {
+        self.arguments_to_context_ids_by_source
+            .get(source)
+            .and_then(|r| r.get(argument))
+            .ok_or_else(|| {
+                SingleFederationError::Internal {
+                    message: "context ID unexpectedly missing for @fromContext argument".to_owned(),
+                }
+                .into()
```

Comment:
should this be `internal_error!()`?

Reply:
Addressed in [259fe72](https://github.com/apollographql/router/pull/6380/commits/259fe723715f9466bbdc6c7a9430766df73e98d7).